assets/js/tutorials/require-destination-tags.js

```diff
@@ -0,0 +1,148 @@
+      // get the wrong starting sequence number.
+      while (true) {
+        try {
+          await api.request("account_info", {account: address, ledger_index: "validated"})
+          break
+        } catch(e) {
+          await new Promise(resolve => setTimeout(resolve, 1000))
+        }
+      }
+    }
+    return {address, secret}
+  }
+
+  // Actual handler for the two buttons in the Send Test Payments block.
+  // Gets the destination tag (or lack thereof) from their data attributes.
+  $(".test-payment").click( async (event) => {
+    const block = $(event.target).closest(".interactive-block")
+    const address = $("#use-address").text()
+    if (!address) {
+      const block = $(event.target).closest(".interactive-block")
+      show_error(block,
+        `Couldn't get a valid address/secret value. Check that the
+         previous steps were completed successfully.`)
+      return
+    }
+
+    block.find(".loader").show()
+    try {
+      const test_sender = await get_test_sender(block)
+      const tx_json = {
+        "TransactionType": "Payment",
+        "Account": test_sender.address,
+        "Amount": "3152021",
+        "Destination": address
+      }
+      const dt = $(event.target).data("dt")
+      if (dt) {
+        tx_json["DestinationTag"] = parseInt(dt)
+      }
+
+      const prepared = await api.prepareTransaction(tx_json)
+      const signed = api.sign(prepared.txJSON, test_sender.secret)
+      console.debug("Submitting test payment", prepared.txJSON)
+      const prelim_result = await api.request("submit",
+                                          {"tx_blob": signed.signedTransaction})
+
+      block.find(".loader").hide()
+      block.find(".output-area").append(`<p>${tx_json.TransactionType}
+        ${prepared.instructions.sequence} ${(dt?"WITH":"WITHOUT")} Dest. Tag:
+        <a href="https://testnet.xrpl.org/transactions/${signed.id}"
+        target="_blank">${prelim_result.engine_result}</a></p>`)
+    } catch(err) {
+      block.find(".loader").hide()
+      show_error(`An error occurred when sending the test payment: ${err}`)
+    }
+
+
+    // SCRAPPED ALT CODE: using the Faucet as a test sender.
+    // // We can use the Faucet to send a payment to our existing address with a
+    // // destination tag, but only if we roll it into an X-address.
+    // const faucet_url = $("#generate-creds-button").data("fauceturl")
+	  // const XCodec = require('xrpl-tagged-address-codec')
+    // const dt = $(event.target).data("dt")
+    // let dest_x_address
+    // if (dt) {
+    //   dest_x_address = XCodec.Encode({ account: address, tag: dt, test: true })
+    // } else {
+    //   dest_x_address = XCodec.Encode({ account: address, test: true })
+    // }
+    // call_faucet(faucet_url, dest_x_address)
+    //
+    // // TODO: monitor our target address and report when we receive the tx from
+    // // the faucet. (including ✅ or ❌ as appropriate)
```

Comment:
I scrapped the code for using the Faucet to send these test transactions for two reasons:

- I needed to pull in another library on top of ripple-lib to get the relevant X-address functions. This was fixed in [ripple-lib 1.9.3](https://github.com/ripple/ripple-lib/blob/develop/HISTORY.md#193-2021-03-16) in response to my questions about this.
- I realized it would be tricky to report the results of a given transaction since the Faucet doesn't tell us anything about the transaction itself. But it might be doable using the account subscriptions pattern I later added to the "Send XRP" tutorial.